spec/support/api_spec_helper.rb

```diff
@@ -16,19 +16,8 @@ module ApiSpecHelper
     "Accept"       => "application/json"
   }
 
-  API_STATUS = Rack::Utils::HTTP_STATUS_CODES.merge(0 => "Network Connection Error")
-
-  def parse_response
-    @code    = response.status
-    @result  = case @code
-               when Rack::Utils.status_code(:no_content)   then {}
-               when Rack::Utils.status_code(:unauthorized) then response.body
-               else JSON.parse(response.body)
-               end
-    @success = response.success?
-    @status  = API_STATUS[@code] || (@success ? Rack::Utils.status_code(:ok) : Rack::Utils.status_code(:bad_request))
-    @message = @result.kind_of?(Hash) ? @result.fetch_path("error", "message").to_s : @result
-    @success
+  def response_hash
+    JSON.parse(response.body)
```

Comment:
@chrisarcand thanks for your comment, and apologies as I only just saw it. I think we are in agreement - I'm not saying I want to avoid making multiple expectations at all costs, but that it is a smell that may benefit some closer attention. In this case, and after giving it more thought, I feel stronger in the don't memoize camp, for a couple of reasons:

1. If a spec is making multiple expectations on this same query, I am sure they could be rewritten to have one.
2.   `response_hash` depends on `response` which can change and which it has no control over. There are several specs in authorization for which memoizing would have an undesirable effect, because they make multiple requests in the same test, and make expectations against the response after each. Memoizing would require us to know more about the state and start messing with it to accommodate tests like these.

Since this has been closed, I'll follow up to address (1) and cc you there if you have any concerns. Thanks!